src/Adapter/Order/CommandHandler/DeleteCartRuleFromOrderHandler.php

```diff
@@ -85,12 +100,74 @@ public function handle(DeleteCartRuleFromOrderCommand $command)
 
         try {
             // Delete Order Cart Rule and update Order
+            $cartRule = new CartRule($orderCartRule->id_cart_rule);
             $orderCartRule->softDelete();
             $cart->removeCartRule($orderCartRule->id_cart_rule);
 
-            $this->orderAmountUpdater->update($order, $cart, $orderCartRule->id_order_invoice);
+            // If cart rule was a gift product we must update an OrderDetail manually
+            $giftOrderDetail = $this->getGiftOrderDetail($order, $cartRule);
+            if (null !== $giftOrderDetail) {
```

Comment:
IMO this should be
```suggestion
            if ($giftOrderDetail instanceof OrderDetail) {
```
especially since you use one of its properties just after

Reply:
Since the function return `?OrderDetail` it can return either an OrderDetail or null
So it's equivalent, it's like checking `$result == true` or `$result != false` 😛

But I'll change it anyway